### ISSUE TITLE
HTML special chars appear encoded in refererence editor

### ISSUE THREAD

**Issue by alex-boehm** [neos/neos-ui#1996]
### Description
![grafik](https://user-images.githubusercontent.com/5336386/42559499-90a311b6-84f4-11e8-93c5-1ed9db36f21b.png)

### Steps to Reproduce

1. Add a node with special chars in title (like &)
2. Create a field of type reference for the node type
3. Select node type with special char

#### Expected behavior

'&' should be displayed as is, not &amp;

#### Actual behavior
Shown as &amp;

[What actually happened]

### Affected Versions

Neos: 4.0.6

UI: 1.2.3

**Comment by skurfuerst** [comment]
Hm, I am a bit unsure where to solve this correctly - to not introduce XSS attacks into the system. Any suggestions @mstruebing @dimaip ?

**Comment by dimaip** [comment]
So yes, we definitively shouldn't just `dangerouslySetInnerHTML`.
As I understand, we should remove tags and decode html entities. Or make CKeditor to not encode html entitles, which is what I think I already tried to do earlier.
